File: src/main/java/com/vc/medium/L901.scala
package com.vc.medium

class StockSpanner() {
  import scala.collection.mutable
  val st = new mutable.Stack[(Int, Int)]()
  def next(price: Int): Int = {
    var res = 0
    while(st.nonEmpty && st.head._1 < price) {
      val e = st.pop
      res += e._2
    }
    st.push((price, res))
    res
  }
}

/**
  * Your StockSpanner object will be instantiated and called as such:
  * var obj = new StockSpanner()
  * var param_1 = obj.next(price)
  */

File: src/main/java/com/vc/medium/L901.java
package com.vc.medium;

import java.util.Stack;

class StockSpanner {
    class Node {
        int price = 0;
        int count = 0;

        Node(int price, int count) {
            this.price = price;
            this.count = count;
        }
    }

    Stack<Node> st;
    public StockSpanner() {
        st = new Stack<>();
    }

    public int next(int price) {
        int count = 1;
        while(!st.isEmpty() && st.peek().price <= price) count += st.pop().count;
        st.push(new Node(price, count));
        return count;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */
